back/spec/models/notification_spec.rb

```diff
@@ -133,6 +133,34 @@
       end
     end
 
+    describe 'project_phase_upcoming' do
+      let(:project) { create(:project_with_current_phase) }
+      let(:other_project) { create(:project_with_current_phase) }
+
+      let(:project_folder) { create(:project_folder, projects: [project]) }
+      let(:other_project_folder) { create(:project_folder, projects: [other_project]) }
+
+      let!(:regular_user) { create(:user) }
+      let!(:moderator_of_project) { create(:project_moderator, projects: [project]) }
+      let!(:moderator_of_other_project) { create(:project_moderator, projects: [other_project]) }
+      let!(:moderator_of_project_folder) { create(:project_folder_moderator, project_folders: [project_folder]) }
+      let!(:moderator_of_other_project_folder) { create(:project_folder_moderator, project_folders: [other_project_folder]) }
+      let!(:admin) { create(:admin) }
+
+      it 'notifies all admins and only relevant moderators' do
```

Comment:
Maybe this test is a little bit redundant. It actually tests just this line `UserRoleService.new.moderators_for(phase)`. And we already have it here `back/spec/services/user_role_service_spec.rb`.

But I understand why you've written it. Probably, because it wasn't clear at first glance what's the difference between audiences of different notifications. But maybe writing some documentation or refactoring some code (that is responsible for the audience) would help even more for this purpose (though it has its own drawbacks).

Reply:
Good point.

In the end, I decided that the test itself acts as documentation, and also could catch regression where the model is changed such that it does not use `user_role_service` in the way it does now.

I also moved the test to the model spec, `project_phase_upcoming_spec.rb`, similar to where we put the tests for `project_phase_started` in #3358.

Reply:
As you moved this spec to another file, are you going to remove it from this file?

Reply:
Yes, my mistake. I am already dealing with it in the other open PR, #3358 .